src/library/scala/collection/Iterable.scala

```diff
@@ -52,3 +57,19 @@ object Iterable extends TraversableFactory[Iterable] {
 
 /** Explicit instantiation of the `Iterable` trait to reduce class file size in subclasses. */
 abstract class AbstractIterable[+A] extends AbstractTraversable[A] with Iterable[A]
+
+/** PairedIterable for functions on paired iterables */
+class PairedIterable[K, V](x: Iterable[(K, V)]) {
+  def reduceByKey(func: (V,V) => V) = {
```

Comment:
Isn't this a special case of a method `reduceToMap[K, V](keyFn: A => K, valFn: A => V, merge: (V, V) => V)`?  Or shouldn't it instead be defined on TraversableOnce so you can do the key/value mapping on the fly (e.g. `xs.iterator.map(a => (k(a), v(a))).reduceByKey(foo)`)?